posthog/storage/team_metadata_cache.py

```diff
@@ -0,0 +1,438 @@
+    "capture_dead_clicks",
+    "autocapture_opt_out",
+    "autocapture_web_vitals_opt_in",
+    "autocapture_web_vitals_allowed_metrics",
+]
+
+
+def _load_team_metadata(team_key: KeyType) -> dict[str, Any] | HyperCacheStoreMissing:
+    """
+    Load full team metadata from the database.
+
+    Args:
+        team_key: Team identifier (can be Team object, API token string, or team ID)
+
+    Returns:
+        Dictionary containing full team metadata, or HyperCacheStoreMissing if team not found
+    """
+    try:
+        with transaction.atomic():
+            # Get the team based on the key type using HyperCache helper
+            team = HyperCache.team_from_key(team_key)
+
+            # Ensure related objects are loaded - more reliable approach
+            if isinstance(team, Team):
+                try:
+                    # Check if relations are accessible without causing a query
+                    _ = team.organization.name
+                    _ = team.project.name
+                except (AttributeError, ObjectDoesNotExist):
+                    # Relations not loaded or don't exist, refetch
+                    team = Team.objects.select_related("organization", "project").get(id=team.id)
+
+            # Build the metadata dictionary with all specified fields
+            metadata = {}
+            for field in TEAM_METADATA_FIELDS:
+                value = getattr(team, field, None)
+
+                # Handle special field types for JSON serialization
+                if field in ["created_at", "updated_at"]:
+                    value = value.isoformat() if value else None
+                elif field == "uuid":
+                    value = str(value) if value else None
+                elif field == "organization_id":
+                    value = str(team.organization_id) if team.organization_id else None
+                elif field == "session_recording_sample_rate":
+                    # Convert Decimal to float for JSON serialization
+                    value = float(value) if value is not None else None
+
+                metadata[field] = value
+
+            # Add computed/related fields - safely access with hasattr checks
+            metadata["organization_name"] = (
+                team.organization.name if hasattr(team, "organization") and team.organization else None
+            )
+            metadata["project_name"] = team.project.name if hasattr(team, "project") and team.project else None
+            metadata["last_updated"] = datetime.now(UTC).isoformat()
+
+            return metadata
+
+    except Team.DoesNotExist:
+        # Log only that a team was not found, without exposing the key
+        logger.warning("Team not found for cache lookup")
+        return HyperCacheStoreMissing()
+
+    except Exception:
+        # Log the error without exposing sensitive team information
+        logger.exception("Error loading team metadata")
+        return HyperCacheStoreMissing()
```

Comment:
In general I think catching `Exception` can be too broad. But in this case, we do want to fail gracefully, but we also don't want to lose any useful infromation for debugging. Perhaps we should do something like:

```suggestion
	except Exception as e:
	      logger.exception(
	          "Error loading team metadata",
	          error_type=type(e).__name__,  # Help debugging
	          team_key_type=type(team_key).__name__,  # Add context
	      )
	      return HyperCacheStoreMissing()
```

And if there _are_ other errors we're expecting to occur, lets catch them above and do the right thing like we do with `Team.DoesNotExist`.